src/lib/y2security/selinux_config.rb

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+module Y2Security
+  # Class for handling SELinux kernel params
+  #
+  # @example Querying the currently configured SELinux mode
+  #   selinux = SelinuxConfig.new
+  #   mode = selinux.mode
+  #   mode.id #=> :permissive
+  #   mode.name #=> "Permisive"
+  #
+  # @example Querying the currently running SELinux mode
+  #   selinux= SelinuxConfig.new
+  #   mode = selinux.running_mode
+  #   mode.id #=> :enforcing
+  #   mode.name #=> "Enforcing"
+  #
+  # @example Enabling SELinux in Permissive mode for next boot
+  #   selinux = SelinuxConfig.new
+  #   selinux.mode = :permissive
+  #   selinux.save #=> true
+  #
+  # @example Disabling SELinux for next boot
+  #   selinux = SelinuxConfig.new
+  #   selinux.mode = :disabled
+  #   selinux.save #=> true
+  #
+  # @example Disabling SELinux for next boot (using nil)
```

Comment:
is there way to simply do nothing? I would like to use it when control file say do not touch it, then I would like to hide UI, but in backend just set it as keep it.

Reply:
@jreidinger see https://github.com/yast/yast-security/pull/83/commits/2b58a20cbe09fb17ea1b4ddf1ac61fd3cf44fded

We can revert it if it is not what you were looking for :)

Reply:
Finally reverted in favor of [`#configurable?`](https://github.com/yast/yast-security/pull/83/commits/2ccd5c23b7bc1f6fd77d6b9e8ff55d199a1b6258)